apps/typegpu-docs/src/content/examples/simple/vaporrave/index.ts

```diff
@@ -0,0 +1,242 @@
+import tgpu from 'typegpu';
+import * as d from 'typegpu/data';
+import * as std from 'typegpu/std';
+import { sdPlane } from '@typegpu/sdf';
+import { perlin3d } from '@typegpu/noise';
+
+import { circles, grid } from './floor.ts';
+import * as c from './constants.ts';
+import { LightRay, Ray } from './types.ts';
+import { getSphere } from './sphere.ts';
+import { rayUnion } from './helpers.ts';
+
+// == INIT ==
+const canvas = document.querySelector('canvas') as HTMLCanvasElement;
+const context = canvas.getContext('webgpu') as GPUCanvasContext;
+const presentationFormat = navigator.gpu.getPreferredCanvasFormat();
+
+const root = await tgpu.init();
+
+context.configure({
+  device: root.device,
+  format: presentationFormat,
+  alphaMode: 'premultiplied',
+});
+
+// == BUFFERS ==
+const floorAngleUniform = root.createUniform(d.f32);
+const sphereAngleUniform = root.createUniform(d.f32);
+const glowIntensityUniform = root.createUniform(
+  d.f32,
+  c.INITIAL_GLOW_INTENSITY,
+);
+const resolutionUniform = root.createUniform(d.vec2f);
+const sphereColorUniform = root.createUniform(d.vec3f, c.initialSphereColor);
+
+let floorSpeed = 1;
```

Comment:
This is just a suggestion, could we start off with a slower floor rotation? This way, the viewer can focus on the sphere and they don't compete for attention.
```suggestion
let floorSpeed = 0.1;
```